### ISSUE TITLE
click and drag inventory management

### ISSUE THREAD

**Issue by Kirv-goldblade** [Yellow-Dog-Man/Resonite-Issues#1011]
### Is your feature request related to a problem? Please describe.

as it stands to move items around your inventory you need to spawn them into the world and re-save then to the location you want them in.

### Describe the solution you'd like

it would be nice to be able to move items around our inventory similar to how we can move slots in a hierarchy, grabbing them and dropping them into a new location

### Describe alternatives you've considered

N/A (situation is binary as the only option for inventory managment currently is the spawn/re-save method)

**Comment by liny-fox** [comment]
Copy and paste would also be nice as its not always directly accessible and might be a few folders away.

**Comment by Kirv-goldblade** [comment]
yeah, i was thinking of being able to hold the item with grab/right click well being able to use primary/left click to navigate the inventory but a copy/cut and paste function would be handy too (even just a paste to inventory button to let us past a resrec url directly into an inventory folder would be a good start (though being able to drag to sort items would be great)

**Comment by Frooxius** [comment]
We don't plan on making incremental updates to the current inventory UI, so this will not be done for the current one.

However the inventory UI is planned to be reworked at some point, so we'd tackle these things with this. I'm not sure if it's going to be drag & drop specifically (it might not fit as well into the control scheme), but you'll be able to move things around in some way.

**Comment by TisFoolish** [comment]
Will the work on the systems for the settings menu be part of the foundation for the new inventory?

**Comment by Frooxius** [comment]
Yes, that's the goal. The data feed and feed views will be used to implement new inventory UI.

**Comment by SlyTheFloof** [comment]
> We don't plan on making incremental updates to the current inventory UI, so this will not be done for the current one.
> 
> However the inventory UI is planned to be reworked at some point, so we'd tackle these things with this. I'm not sure if it's going to be drag & drop specifically (it might not fit as well into the control scheme), but you'll be able to move things around in some way.

That's fantastic to hear we may have a change to move things around more simply then having to rez out objects and wait for syncing as it would already be in the user inventory just path changes in due time
